src/main/java/ch/njol/skript/expressions/ExprFuseDuration.java

```diff
@@ -0,0 +1,157 @@
+		}
+		return null;
+	}
+
+	private Timespan getTimespan(int ticks) {
+		if (ticks < 0) {
+			ticks = 0;
+		}
+		return new Timespan(Timespan.TimePeriod.TICK, ticks);
+	}
+
+
+	@Override
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE || mode == ChangeMode.DELETE || mode == ChangeMode.RESET) {
+			return CollectionUtils.array(Timespan.class);
+		}
+		return null;
+	}
+
+	@Override
+	public void change(Event event, Object @NotNull [] delta, ChangeMode mode) {
+		for (Entity entity : getExpr().getArray(event)) {
+			if (entity instanceof TNTPrimed) {
+				changeTNTFuseTicks((TNTPrimed) entity, delta, mode);
+			} else if (entity instanceof Creeper) {
+				changeCreeperFuseTicks((Creeper) entity, delta, mode);
+			} else if (entity instanceof ExplosiveMinecart) {
+				changeMinecartFuseTicks((ExplosiveMinecart) entity, delta, mode);
+			}
+		}
+	}
+
+	private void changeTNTFuseTicks(TNTPrimed tnt, Object[] delta, ChangeMode mode) {
+		int currentTicks = tnt.getFuseTicks();
+		int newTicks = calculateNewTicks(tnt, currentTicks, delta, mode);
+		tnt.setFuseTicks(newTicks);
+	}
+
+	private void changeCreeperFuseTicks(Creeper creeper, Object[] delta, ChangeMode mode) {
+		if (max) {
+			int currentMaxTicks = creeper.getMaxFuseTicks();
+			int newMaxTicks = calculateNewTicks(creeper, currentMaxTicks, delta, mode);
+			creeper.setMaxFuseTicks(Math.max(newMaxTicks, 0));
+		}
+
+		int currentTicks = creeper.getFuseTicks();
+		int maxTicks = creeper.getMaxFuseTicks();
+		int newTicks = calculateNewTicks(creeper, currentTicks, delta, mode);
+
+		creeper.setFuseTicks(Math.max(Math.min(newTicks, maxTicks), 0)); // Ensure non-negative value
+	}
+
+	private void changeMinecartFuseTicks(ExplosiveMinecart minecart, Object[] delta, ChangeMode mode) {
+		int currentTicks = minecart.getFuseTicks();
+		int newTicks = calculateNewTicks(minecart, currentTicks, delta, mode);
+		minecart.setFuseTicks(newTicks);
+	}
+
+	private int calculateNewTicks(Entity entity, int currentTicks, Object[] delta, ChangeMode mode) {
+		long deltaTicks = delta != null && delta.length > 0 ? ((Timespan) delta[0]).getAs(Timespan.TimePeriod.TICK) : 0;
+		long newTicks = switch (mode) {
+			case SET -> deltaTicks;
+			case ADD -> (long) currentTicks + deltaTicks;
+			case REMOVE -> (long) currentTicks - deltaTicks;
+			case DELETE -> 0;
+			case RESET -> getDefaultTicks(entity);
+			default -> currentTicks;
+		};
+
+		// check for overflow
+		if (newTicks > Integer.MAX_VALUE) {
+			newTicks = Integer.MAX_VALUE;
+		} else if (newTicks < 0) {
+			newTicks = 0;
+		}
+
+		return (int) newTicks;
+	}
+
+	private int getDefaultTicks(Entity entity) {
+		if (entity instanceof TNTPrimed) {
+			return 80;
+		} else if (entity instanceof Creeper) {
+			return 30;
+		} else if (entity instanceof ExplosiveMinecart) {
+			return 80;
+		}
+		return 80; // fallback
```

Comment:
So I'm slightly concerned that these are hardcoded and could potentially change, but I can't find anywhere we could get these internally (easily) so I don't think we have a way around that, we'll just have to keep an eye on it in case any get a gamerule or something later.